cpp/src/jit/type.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <string>
+
+namespace cudf {
+namespace jit {
+
+struct get_data_ptr_functor {
+  
+  /**
+   * @brief Gets the data pointer from a column_view
+   */
+  template <typename T>
+  std::enable_if_t<is_fixed_width<T>(), const void *>
+  operator()(column_view const& view) {
+    return static_cast<const void*>(view.template data<T>());
+  }
+
+  // TODO: both the failing operators can be combined into single template
+  template <typename T>
+  std::enable_if_t<not is_fixed_width<T>(), const void *>
+  operator()(column_view const& view) {
+    CUDF_FAIL("Invalid data type for JIT operation");
+  }
+
+  /**
+   * @brief Gets the data pointer from a scalar
+   */
+  template <typename T>
+  std::enable_if_t<is_fixed_width<T>(), const void *>
+  operator()(scalar const& s) {
+    using ScalarType = experimental::scalar_type_t<T>;
+    auto s1 = static_cast<ScalarType const*>(&s);
+    return static_cast<const void*>(s1->data());
+  }
```

Comment:
missing empty line between the definitions of two operators.

Reply:
Addressed in https://github.com/rapidsai/cudf/pull/3507/commits/ca001cf62e4e41859dd7ad75e0c5ff8ea53eebe0.